torch/export/exported_program.py

```diff
@@ -173,8 +173,45 @@ def _register_cia_to_meta(*args, **kwargs):
 ]
 
 
+# Our strategy for deciding if we can preserve a op is following:
+# 1. The op should be known statically that it is functional
+# 2. If it is maybe aliasing, we decompose because we must know if an op
+#    is mutating or aliasing.
+# TODO (tmanlaibaatar) make this utility function and share it with functional_tensor
+# decomp part. (https://github.com/pytorch/pytorch/issues/129431)
+def _assert_valid_to_preserve(op_overload):
+    if op_overload in FunctionalTensor.maybe_aliasing_or_mutating_ops:
+        raise RuntimeError(
+            f"We can't detect {op_overload} as a functional op statically, so we can't preserve it"
+        )
+    if op_overload in FunctionalTensor.metadata_fns:
```

Comment:
(oh this is existing code being moved - comment is still valid though, although if we want to take any action it can probably be independent from this PR)

Reply:
Yeah i have a separate PR for that. The goal of this check is to error if any op user want to preserve has some unwanted behaviours like maybe aliasing, or just straight up mutating ops/